dds/idl/typeobject_generator.cpp

```diff
@@ -11,339 +11,1593 @@
 #include "utl_identifier.h"
 
 #include "topic_keys.h"
+#include "dds_visitor.h"
+#include "dds/DCPS/Hash.h"
 
 using std::string;
 using namespace AstTypeClassification;
 
 namespace {
 
+OpenDDS::XTypes::TypeFlag
+extensibility_to_type_flag(ExtensibilityKind exten)
+{
+  switch (exten) {
+  case extensibilitykind_final:
+    return OpenDDS::XTypes::IS_FINAL;
+  case extensibilitykind_appendable:
+    return OpenDDS::XTypes::IS_APPENDABLE;
+  case extensibilitykind_mutable:
+    return OpenDDS::XTypes::IS_MUTABLE;
+  }
+
+  return 0;
+}
+
+OpenDDS::XTypes::MemberFlag
+try_construct_to_member_flag(TryConstructFailAction trycon)
+{
+  switch (trycon) {
+  case tryconstructfailaction_discard:
+    return OpenDDS::XTypes::TryConstructDiscardValue;
+  case tryconstructfailaction_use_default:
+    return OpenDDS::XTypes::TryConstructUseDefaultValue;
+  case tryconstructfailaction_trim:
+    return OpenDDS::XTypes::TryConstructTrimValue;
+  }
+
+  return 0;
+}
+
+ACE_CDR::Long
+to_long(const AST_Expression::AST_ExprValue& ev)
+{
+  switch (ev.et) {
+  case AST_Expression::EV_octet:
+    return ev.u.oval;
+  case AST_Expression::EV_short:
+    return ev.u.sval;
+  case AST_Expression::EV_ushort:
+    return ev.u.usval;
+  case AST_Expression::EV_long:
+    return ev.u.lval;
+  case AST_Expression::EV_ulong:
+    return ev.u.ulval;
+  case AST_Expression::EV_longlong:
+    return ev.u.llval;
```

Comment:
What are the implications of narrowing/truncation here?

Reply:
This function is use for computing case labels so it would have to be a type that can be used as a union discriminator.
I'll change the code to abort for types that cannot be represented by a long.

Reply:
Seems like a bug in the spec that cases for `long long` discriminators can't be represented

Reply:
Agreed.  The relevant line from the spec is `typedef sequence<long> UnionCaseLabelSeq`.

Reply:
New spec issue created: DDSXTY14-36, you can add a comment here with that identifier.